src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/McpServer-CSharp.csproj.in

```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net10.0</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    
+    <!-- Set up the NuGet package to be an MCP server -->
+    <PackAsTool>true</PackAsTool>
+    <PackageType>McpServer</PackageType>
+
```

Comment:
If you want to recommend RID-specificity in the MCP servers that users make, check out [my comment on your linked PR](https://github.com/dotnet/docs/pull/47007/files#r2175570614). I would consider defaulting a set of `RuntimeIdentifiers` here for users to start from, and you were considering making them `PublishSelfContained=true` for p6 and `PublishTrimmed=true` for p7.

Reply:
Added https://github.com/dotnet/extensions/issues/6560 for tracking, for P7.